youtube_dl/extractor/kakao.py

```diff
@@ -0,0 +1,140 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none,
+)
+import datetime
+
+
+class KakaoIE(InfoExtractor):
+    _VALID_URL = r'https?://tv.kakao.com/channel/(?P<channel>\d+)/cliplink/(?P<id>\d+)'
+    IE_NAME = 'kakao.com'
+
+    _TESTS = [{
+        'url': 'http://tv.kakao.com/channel/2671005/cliplink/301965083',
+        'md5': '702b2fbdeb51ad82f5c904e8c0766340',
+        'info_dict': {
+            'id': '301965083',
+            'ext': 'mp4',
+            'title': '乃木坂46 バナナマン 「3期生紹介コーナーが始動！顔高低差GPも！」 『乃木坂工事中』',
+            'uploader_id': 2671005,
+            'uploader': '그랑그랑이',
+            'upload_date': '20170227'
+        }
+    }, {
+        'url': 'http://tv.kakao.com/channel/2653210/cliplink/300103180',
+        'md5': 'a8917742069a4dd442516b86e7d66529',
+        'info_dict': {
+            'id': '300103180',
+            'ext': 'mp4',
+            'description': '러블리즈 - Destiny (나의 지구) (Lovelyz - Destiny)\r\n\r\n[쇼! 음악중심] 20160611, 507회',
+            'title': '러블리즈 - Destiny (나의 지구) (Lovelyz - Destiny)',
+            'uploader_id': 2653210,
+            'uploader': '쇼 음악중심',
+            'upload_date': '20170129'
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        # Player URL, to be used in Referer header
+        player_url = 'http://tv.kakao.com/embed/player/cliplink/' + video_id + \
+            '?service=kakao_tv&autoplay=1&profile=HIGH&wmode=transparent'
+        player_header = {'Referer': player_url}
+
+        # Request Impress, which contains video information
+        impress = self._download_json(
+            'http://tv.kakao.com/api/v1/ft/cliplinks/%s/impress' % video_id,
+            video_id, 'Downloading video info',
+            query={
+                'player': 'monet_html5',
+                'referer': url,
+                'uuid': '',
+                'service': 'kakao_tv',
+                'section': '',
+                'dteType': 'PC',
+                'fields': 'clipLink,clip,channel,hasPlusFriend,-service,-tagList'
+            }, headers=player_header)
+
+        clipLink = impress['clipLink']
+
+        # Now we request Raw, which contains infos about video files.
+        tid = impress.get('tid', '')
+        raw = self._download_json(
+            'http://tv.kakao.com/api/v1/ft/cliplinks/%s/raw' % video_id,
+            video_id, 'Downloading video formats info',
+            query={
+                'player': 'monet_html5',
+                'referer': url,
+                'uuid': '',
+                'service': 'kakao_tv',
+                'section': '',
```

Comment:
Though they really are, the parameters were hard-coded in the JavaScript. In other words, they need not be any same, or share some of the parameters.
Would it be better to make an (unexplainable) dict of the unknowingly same parameters and append some new parameters for each query?